apis/submariner/v1alpha1/submariner_types.go

```diff
@@ -102,7 +110,20 @@ type HealthCheckSpec struct {
 	MaxPacketLossCount uint64 `json:"maxPacketLossCount,omitempty"`
 }
 
-const DefaultColorCode = "blue"
+const (
+	DefaultColorCode       = "blue"
+	DefaultKubernetesType  = "k8s"
+	KubernetesTypeK8s      = "k8s"
+	KubernetesTypeOCP      = "ocp"
+	KubernetesTypeEKS      = "eks"
+	KubernetesTypeAKS      = "aks"
+	KubernetesTypeGKE      = "gke"
+	CloudProviderKind      = "kind"
+	CloudProviderAWS       = "aws"
+	CloudProviderGCP       = "gcp"
+	CloudProviderAzure     = "azure"
+	CloudProviderOpenstack = "openstack"
+)
```

Comment:
```suggestion
type KubernetesType string
type CloudProvider string

const (
	DefaultColorCode                     = "blue"
	K8s                   KubernetesType = "k8s"
	OCP                                  = "ocp"
	EKS                                  = "eks"
	AKS                                  = "aks"
	GKE                                  = "gke"
	DefaultKubernetesType                = K8s
	Kind                  CloudProvider  = "kind"
	AWS                                  = "aws"
	GCP                                  = "gcp"
	Azure                                = "azure"
	Openstack                            = "openstack"
)
```

Perhaps something like this?

Reply:
I considered this, but thought using more descriptive constant name will make it easy to understand what they're used for.

Reply:
We also have "Kind" kubernetes type that we may want to add.

And... for cloud provider ir could also have vSphere, or even unknown?

Reply:
Can you put it in it's own package? Then it's as explicit as you can get.

> We also have "Kind" kubernetes type that we may want to add.

Is it though? I think K8s "type" and Kind "cloud" are enough to understand its on KIND..

> 
> And... for cloud provider ir could also have vSphere, or even unknown?